demo-app/src/androidTestMock/java/com/worldpay/access/checkout/sample/NewDiscoveryIntegrationTest.kt

```diff
@@ -0,0 +1,69 @@
+package com.worldpay.access.checkout.sample
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import com.worldpay.access.checkout.client.session.model.SessionType.CARD
+import com.worldpay.access.checkout.client.session.model.SessionType.CVC
+import com.worldpay.access.checkout.sample.card.standard.testutil.NewCardFragmentTestUtils
+import com.worldpay.access.checkout.sample.cvc.NewCvcFragmentTestUtils
+import com.worldpay.access.checkout.sample.stub.RootResourseMockStub.simulateRootResourceTemporaryServerError
+import com.worldpay.access.checkout.sample.testutil.UITestUtils.navigateTo
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class NewDiscoveryIntegrationTest {
+    // WIP
+
+    private val amexCard = "343434343434343"
+    private val amexCvc = "1234"
+    private val expiryDate = "1299"
+
+    @get:Rule
+    var activityTestRule = ActivityScenarioRule(MainActivity::class.java)
+
+    @Before
+    fun setup() {
+        simulateRootResourceTemporaryServerError()
+    }
+
+    @Test
+    fun shouldRetryDiscoveryAndReturnSuccessfulResponse_whenDiscoveryFailsFirstTime_cardFlow() {
```

Comment:
`successfulResponse` is misleading because it seems to pertain to the discovery. Can we rename `AndReturnSuccessfulResponse` maybe `AndSuccessfullyCreatesSession`?